kolibri/plugins/management/assets/src/vue/modal/index.vue

```diff
@@ -1,66 +1,115 @@
 <template>
 
-  <div class="modal-root" v-on:keyup.esc="closeModal">
-    <div class="modal" v-if="showModal" transition="modal">
-      <div class="modal-wrapper">
-        <div class="modal-container">
-          <button @click="closeModal" class="close-btn">
-            <svg src="../icons/close.svg"></svg>
-            <span class="visuallyhidden">Close</span>
-          </button>
-          <div class="modal-header">
-            <slot name="header">
-              Kolibri
-            </slot>
-          </div>
-          <div class="modal-body">
-            <slot name="body">
-              disappear in oblivion..
-            </slot>
-          </div>
-          <div class="modal-footer">
-            <slot name="footer">
-              <button @click="closeModal" class="close-btn">OK</button>
-            </slot>
-          </div>
-        </div>
-      </div>
-    </div>
-
-    <div @click="openModal">
-    <!-- wrap this named slot so that the openModal method logic is encapsulated inside this modal component, but the parent component can pass anything to this slot for styling purpose -->
-      <slot name="openbtn">
-        <button>{{ btntext }}</button>
+  <!-- Accessibility properties for the overlay -->
+
+  <!-- Aria-Hidden and TabIndex in .modal might not be necessary because of conditional rendering -->
+  <!-- mostly there in case we switch to v-show -->
+  <div class="modal-overlay"
+    v-if="visible"
+    @keyup.esc="closeModal"
+    @click="bgClick($event)"
+    v-el:modal-overlay
+    id="modal-window"
+    role="dialog"
+    :aria-hidden="(!visible).toString()"
+    :aria-labelledby="title ? 'modal-title' : null"
+    :aria-label="title ? null : 'popup-modal'">
+
+    <div class="modal" v-el:modal :tabindex="visible ? '0' : '-1'" transition="modal">
+      <!-- Close Button -->
+      <button aria-label="close" @click="closeModal" class="btn-close">
+        <svg src="../icons/close.svg"></svg>
+      </button>
+
+      <!-- Modal Title -->
+      <!-- Not mandatory, but if available, names the modal according aria-labels -->
+      <h1 v-if="title" class="title" id="modal-title">{{title}}</h1>
+
+      <!-- Modal Content -->
+      <slot name="body" class="modal-content" id="modal-holder" role="document">
+        <p>
+          To populate, add <code>slot="body"</code> to the HTML element you want to fill here.
+        </p>
       </slot>
+
     </div>
   </div>
 
+  <!-- Think this might be more appropriate as a boolean, or case-by-case -->
+  <div v-if="btntext" @click="openModal">
```

Comment:
I think it might be better to separate out the modal and the button that launches the modal into different components.